kustomize/commands/edit/add/addbuildmetadata.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addBuildMetadataOptions struct {
+	buildMetadataOptions []string
+}
+
+// newCmdAddBuildMetadata adds options to the kustomization's buildMetada field.
+func newCmdAddBuildMetadata(fSys filesys.FileSystem) *cobra.Command {
+	var o addBuildMetadataOptions
+
+	cmd := &cobra.Command{
+		Use:   "buildmetadata",
+		Short: "Adds one or more buildMetadata options to the kustomization.yaml in the current directory",
+		Example: `
+		add buildmetadata {option1},{option2}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+			return o.RunAddBuildMetadata(fSys)
+		},
+	}
+	return cmd
+}
+
+// Validate validates addBuildMetadata command.
+func (o *addBuildMetadataOptions) Validate(args []string) error {
+	if len(args) != 1 {
+		return errors.New("must specify a buildMetadata option")
+	}
+	o.buildMetadataOptions = strings.Split(args[0], ",")
+	for _, opt := range o.buildMetadataOptions {
+		if !kustfile.StringInSlice(opt, types.BuildMetadataOptions) {
+			return fmt.Errorf("invalid buildMetadata option: %s", opt)
+		}
+	}
+	return nil
+}
+
+// RunAddBuildMetadata runs addBuildMetadata command (do real work).
+func (o *addBuildMetadataOptions) RunAddBuildMetadata(fSys filesys.FileSystem) error {
+	mf, err := kustfile.NewKustomizationFile(fSys)
+	if err != nil {
+		return err
+	}
+	m, err := mf.Read()
+	if err != nil {
+		return err
+	}
+	for _, opt := range o.buildMetadataOptions {
+		if kustfile.StringInSlice(opt, m.BuildMetadata) {
```

Comment:
Would it be more handy to ignore this error but replacing the entire buildmetadata fields? For example,  my existing options has `OriginAnnotations` and `TransformerAnnotations`, now I want to only keep `OriginAnnotations`. there seems to be no simple way to do that via `kustomize edit` cmd. It would be handy if users can run `kustomize edit add buildmetadata OriginAnnotations` to achieve the goal.  WDYT? 

Does kustomize have another field that supports multiple builtin values?

Reply:
If we add `kustomize edit remove buildMetadata transformerAnnotations`, would that achieve the goal? I think replacing the entire buildmetadata field might be better under the `set` subcommand, e.g. `kustomize edit set buildMetadata originAnnotations` should set the whole field. IMO we can have all of `set`, `add`, and `remove`. WDYT? 

> Does kustomize have another field that supports multiple builtin values?

No, it doesn't.

Reply:
Didn't aware of `kustomize edit set`. Your proposal sounds good!

Reply:
done, added `kustomize edit set buildmetadata` and `kustomize edit remove buildmetadata` to this PR.